### PR TITLE
doc: add some sections related with Github issues.

### LINKED ISSUE #481: How we handle Github issues
**Is your feature request related to a problem? Please describe.**

Improve the CONTRIBUTING document by adding detailed information on how we handle Github issues with no description or with insufficient information.

**Describe the solution you'd like**

In both cases, the author of the issue will be notified through a comment to add a description and/or to improve it. 
If the Github issue is labeled with "Stale" (no activity for more than 60 days) and the information is still missing then the issue will be automatically closed.

**Describe alternatives you've considered**

To allow the bot to close it in case of a missing description but we decided to give a chase to the issue author 😉

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by fargozhu**
## Description

I am adding some sections related to Github issues and how we are handling (right or wrong) them.

fixes #481 

## Motivation and Context

There's some confusion around labels and queues meaning and also where to start.

## How Has This Been Tested?
N/A

## Screenshots (if appropriate):
N/A

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code has security implications.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,32 +2,113 @@
...
 
-Before kicking the ball, please **do NOT report security vulnerabilities in public issues**! Instead, report with the CloudFoundry Foundation team first <security@cloudfoundry.org> and give us some time to fix it in a fashion time matter before disclosing it to the public. For more information check the CloudFoundry Security [page](https://www.cloudfoundry.org/security/).
+Before kicking the ball, please **do NOT report security vulnerabilities in public issues**! Instead, report with the
+CloudFoundry Foundation team first <security@cloudfoundry.org> and give us some time to fix it in a fashion time matter
+before disclosing it to the public. 
+For more information check the CloudFoundry Security [page](https://www.cloudfoundry.org/security/).
 
-When contributing to this repository, please first discuss the changes through an existent Github issue with the main contributers.
+When contributing to this repository, please first discuss the changes through an existent Github issue with the main
+contributers.
```
Comment (gaktive) [inline CONTRIBUTING.md]: Typo: contributors :)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,32 +2,113 @@
...
 
 ## Code review process
 
 The core team looks to Pull Requests on a regular basis and in a best effort.
 
+## How are Github issues handled
+
+### Queues
+
+1. icebox: contain news issues and issues that are waiting for grooming and not read to start working yet. Some issues
```
Comment (gaktive) [inline CONTRIBUTING.md]: Should items like `icebox` and `todo` be capitalized to match the case of the project columns?

Comment (gaktive) [reply CONTRIBUTING.md]: Leaving as is in case they're variables.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,31 +2,112 @@
...
-The following is a set of guidelines for contributing to KubeCF and its modules. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+The following is a set of guidelines for contributing to KubeCF and its modules. These are mostly guidelines, not rules.
+Use your best judgment and feel free to propose changes to this document in a pull request.
 
 ## How to contribute
 
-Before kicking the ball, please **do NOT report security vulnerabilities in public issues**! Instead, report with the CloudFoundry Foundation team first <security@cloudfoundry.org> and give us some time to fix it in a fashion time matter before disclosing it to the public. For more information check the CloudFoundry Security [page](https://www.cloudfoundry.org/security/).
+Before kicking the ball, please **do NOT report security vulnerabilities in public issues**! Instead, please report with the
+CloudFoundry Foundation team first <security@cloudfoundry.org> and give us some time to fix it in a fashion time matter
+before disclosing it to the public.
```
Comment (satadruroy) [inline CONTRIBUTING.md]: Suggest changing it to:

"please report to the CloudFoundry foundation team and we will fix the issues as quickly as we can before the the details are shared with the public."

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -2,31 +2,112 @@
...
+For more information, check the CloudFoundry Security [page](https://www.cloudfoundry.org/security/).
 
-When contributing to this repository, please first discuss the changes through an existent Github issue with the main contributers.
+When contributing to this repository, please first discuss the changes through an existing Github issue with the main
+contributors.
 
 ## Issues tracker
 
-We're commited to have short and accurate templates for different type issues, in order to gather the required information to start a conversation.
+We're commited to have short and accurate templates for different type issues in order to gather the required
```
Comment (satadruroy) [inline CONTRIBUTING.md]: different type **of ** issues...

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -2,31 +2,112 @@
...
+information to start a conversation.
 
 Once again, feel free to contribute to improve them by opening a pull request.
 
 ### How to report a bug
 
-Start by searching [bugs][1] for some similiar issues and/or extra information. If your search doesn't bring you any help then open an issue by selecting the issue type "Bug" and fill the template accurately as possible.
+Start by searching [bugs][1] for some similiar issues and/or extra information. If your search doesn't bring you any
+help, then open an issue by selecting the issue type "Bug" and fill the template accurately as possible.
```
Comment (satadruroy) [inline CONTRIBUTING.md]: If you are unable to find any relevant information from your search...

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -2,31 +2,112 @@
...
 
 ### How to suggest a feature/enhancement
 
-If you find yourself wishing for a feature/enhancement that does not exist yet in KubeCF start by run a quick [search][2] - you may not be alone! If you ain't any luck, then go ahead and open a new issue by selecting the "Feature" issue type and answer some needed questions.
+If you find yourself wishing for a feature/enhancement that does not exist yet in KubeCF, start by run a
```
Comment (satadruroy) [inline CONTRIBUTING.md]: If you would like to request for a feature that does not yet exist in KubeCF...

Comment (andreas-kupries) [reply CONTRIBUTING.md]: > If you would like to request  a feature that does not yet exist in KubeCF...
(removed `for`)

File: CONTRIBUTING.md (hunk 3 above)
Comment (andreas-kupries) [reply CONTRIBUTING.md]: `the the details` -> `the details`

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -2,31 +2,112 @@
...
-The core team looks to Pull Requests on a regular basis and in a best effort.
+The core team looks at Pull Requests on a regular basis on a best-effort basis.
+
+## How are Github issues handled
+
+### Queues
+
+1. icebox: contain new issues along with issues that are waiting for grooming and not read to start working yet. Some issues
+may be closed or rejected.
+
```
Comment (satadruroy) [inline CONTRIBUTING.md]: Suggest rewording to:

"New issues that require further grooming. Note that some issues may be closed or rejected.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -2,31 +2,112 @@
...
+3. in progress: issues being implemented.
+
+4. under review: issues waiting to be reviewed.
+
+5. done: all issues closed.
+
+### Labels
+
+The labels are grouping in: **priority**, **status** and **type**. There are some individual labels like
+**good first issues** or **help wanted** that can be used but are not tracked from a project management perspecitve.
```
Comment (satadruroy) [inline CONTRIBUTING.md]: Issues are labeled on their **priority**, **status** and **type**.

File: CONTRIBUTING.md (hunk 7 above)
Comment (andreas-kupries) [reply CONTRIBUTING.md]: IMHO "Contains new issues that ..." (note the trailing `s` on `Contains`, vs original, and capitalization).

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -2,31 +2,112 @@
...
+4. **Validation**: issue has enough information but it requires a discussion before accepted.
+
+[todo] more about the labels grouping in future versions of the document.
+
+
+### Creating New Issues
+
+The issue created on Github will be automatically assigned to the product owner, who will then evaluate the issue from an
+information-gathering perspective.
+
```
Comment (satadruroy) [inline CONTRIBUTING.md]: Suggest leaving out information-gathering perspective.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -2,31 +2,112 @@
...
+If an issue doesn't contain any description and/or if the same is not enough to assess, then more information will be
+requested to the issue author.
+
```
Comment (satadruroy) [inline CONTRIBUTING.md]: If the issue does not contain sufficient information the author will be requested to provide additional details.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -2,31 +2,112 @@
...
+In case of an issue gets labeled "Stale" (more than 60 days without any activity), the product owner will:
+
+1. close the issue in case of lack of information.
```
Comment (satadruroy) [inline CONTRIBUTING.md]: In case an issue goes "stale"...

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -2,31 +2,112 @@
...
+2. close the issue if it's no longer relevant.
+3. change labels if it's still relevant and information is sufficient.
+
+### Where to start
+
+All contributions are welcome but the best place to pick an issue to work on is from the _todo_ queue that is ordered by
+priority. Before you move it to the _in progess_ queue, make sure that the story is clear. If not, drop comments on the
+issue.
```
Comment (satadruroy) [inline CONTRIBUTING.md]: If not, add your comments and or questions to the issue.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -2,31 +2,112 @@
...
+
+### The flow
+
+The issue flow is always to the right that means if you pick an issue from the _todo_ queue, you should work on it until
+it's ready to be reviewed by a peer.
+
+If you need to move the issue back to a previous queue, then please leave a clear comment stating the reasons so the
+project core contributors can refine it.
```
Comment (satadruroy) [inline CONTRIBUTING.md]: s/queue/column?

File: CONTRIBUTING.md (hunk 11 above)
Comment (andreas-kupries) [reply CONTRIBUTING.md]: `In case an` -> `When an`

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -2,31 +2,112 @@
...
+
+1. pick an issue from the _todo_ queue and move it to the in progress.
+2. add your name to the assign field and other members name if they are pairing with you.
```
Comment (satadruroy) [inline CONTRIBUTING.md]: s/assign/assignee ?

File: CONTRIBUTING.md (hunk 12 above)
Comment (andreas-kupries) [reply CONTRIBUTING.md]: `and or` -> `and/or`

File: CONTRIBUTING.md (hunk 14 above)
Comment (andreas-kupries) [reply CONTRIBUTING.md]: `member's names` (possessive `'s`, and plural `s`)

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -2,31 +2,109 @@
 
 First, thanks for taking the time to contribute to the project! Your support is highly appreciated - kudos!
 
-The following is a set of guidelines for contributing to KubeCF and its modules. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+The following is a set of guidelines for contributing to KubeCF and its modules. These are mostly guidelines, not rules.
```
Comment (mudler) [inline CONTRIBUTING.md]: Please, also update https://github.com/cloudfoundry-incubator/kubecf-docs/blob/master/content/en/docs/Contribution%20guidelines/_index.md as you see fit

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -2,31 +2,109 @@
...
+Use your best judgment and feel free to propose changes to this document in a pull request.
 
 ## How to contribute
 
-Before kicking the ball, please **do NOT report security vulnerabilities in public issues**! Instead, report with the CloudFoundry Foundation team first <security@cloudfoundry.org> and give us some time to fix it in a fashion time matter before disclosing it to the public. For more information check the CloudFoundry Security [page](https://www.cloudfoundry.org/security/).
+Please report to the CloudFoundry foundation <security@cloudfoundry.org> and we will fix the issues as quickly as we can before the the details are shared with the public
```
Comment (mudler) [inline CONTRIBUTING.md]: I think it misses the context of "Security issues". Otherwise it can be misunderstood

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -2,31 +2,109 @@
...
+
+### Where to start
+
+All contributions are welcome but the best place to pick an issue to work on is from the _todo_ column that is ordered by
+priority. Before you move it to the _in progess_ column, make sure that the story is clear. If not, add your comments and/or 
+questions to the issue.
+
+### How do we estimate
+
+We use t-shirt sizes (XS, S, M, L, XL, XXL)  to determine the Github issue complexity and not the implementation effort, so we can determine if the issue needs to be split in multiple ones and/or if it requires a more extended brainstorm to determine the feasibility of the new feature.
```
Comment (mudler) [inline CONTRIBUTING.md]: I don't see any relation with the aforementioned labels, are those going to be new ones or there is an implicit size relation?

Comment (fargozhu) [reply CONTRIBUTING.md]: new ones

File: CONTRIBUTING.md (hunk 16 above)
Comment (fargozhu) [reply CONTRIBUTING.md]: Security vulnerabilities should be enough.

Comment (mudler) [reply CONTRIBUTING.md]: wait.. wasn't that removed?

**Pull request merged**